consul/src/main/scala/io/buoyant/consul/v1.scala

```diff
@@ -44,7 +49,7 @@ package object v1 {
     Try(mapper.readValue[T](bytes, begin, end - begin))
   }
 
-  object Api {
+  object CatalogApi {
```

Comment:
I don't think we would factor up `apply`, but definitely `infiniteRetryFilter` and `retryClient` could be factored up.

```
trait Api {
  ...
}

object CatalogApi extends Api {
  ...
}

object KvApi extends Api {
  ...
}
```